### DISCUSSION TITLE
SilentMode: "Power Reduction" & Other Notes (CS6800i/WLW176i)

### DISCUSSION THREAD

**Discussion by MarkusMit** [emsesp/EMS-ESP32#2215] [Ideas]
Hi, I made a few observations regarding the Silent Mode ("low-noise operation", de: "Geräuscharmer Betrieb") of my Bosch heat pump CS6800i and the entities available in EMS-ESP32.

## Power Reduction

It seems my heat pump has a setting for Silent Mode ("low-noise operation", de: "Geräuscharmer Betrieb") which is not yet supported by EMS-ESP32, or I was unable to find it: 
I can configure the level of "power reduction" (de: "Leistungsreduktion") between 30 % and 60 % in steps of 10 %. 
![grafik](https://github.com/user-attachments/assets/3a61d3b7-fe60-4e77-ba58-ac6e642f8dc1)

```log
15:38:16.660 TRACE 1142: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 06 (offset 64)
15:38:21.802 TRACE 1169: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 05 (offset 64)
15:38:29.923 TRACE 1240: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 04 (offset 64)
15:38:30.364 TRACE 1243: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 04 (offset 64)
15:38:51.439 TRACE 1338: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 05 (offset 64)
15:38:51.511 TRACE 1339: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 05 (offset 64)
```
[ems-esp-log-silent-leistung.txt](https://github.com/user-attachments/files/17791170/ems-esp-log-silent-leistung.txt)

## Silent Mode Start/End time

Before activating the _silent mode_ and changing both, the start and end time, from their default values, the entities were not visible in the EMS-ESP32 Web UI (and also not Home Assistant ("not available")). 
Could this be changed that the entities "exist" already before first change?

Furthermore, would it be possible to display the times with _time formatting_ (`hh:mm`) instead of `<hh> hours [<mm> minutes]` in the Web UI, and also send them as _time of day_ enties to Home Assistant (if that's possible) instead of minutes?
![grafik](https://github.com/user-attachments/assets/844c127f-da1f-46b7-ab9e-2d809e54dabc)

## Minimum Temperature 

On my heat pump the minimum temperature for the silent mode (will be deactivated below that) can configured from (+20 °C `14` to -30 °C `E2`), but it  it can also be _deactivated_ (value `E1`).
Would it be possible to support that "inactive" mode also somehow more explicitly in EMS-ESP32 (and Home Assistant ;-) )?
```log
16:58:08.967 TRACE 3320: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: E2 (offset 11)
16:58:09.027 TRACE 3321: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: E2 (offset 11)
16:58:15.135 TRACE 3348: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: E1 (offset 11)
16:58:15.205 TRACE 3349: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: E1 (offset 11)
16:58:18.728 TRACE 3367: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: E2 (offset 11)
16:58:18.792 TRACE 3368: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: E2 (offset 11)
16:58:27.706 TRACE 3399: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: F5 (offset 11)
16:58:27.729 TRACE 3400: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: F5 (offset 11)
16:58:32.868 TRACE 3470: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: F6 (offset 11)
16:58:32.931 TRACE 3471: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: F6 (offset 11)
16:58:41.884 TRACE 3518: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 14 (offset 11)
16:58:41.956 TRACE 3519: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 14 (offset 11)
16:58:46.784 TRACE 3522: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 13 (offset 11)
16:58:46.844 TRACE 3523: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 13 (offset 11)
16:58:54.896 TRACE 3530: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 00 (offset 11)
16:58:54.966 TRACE 3531: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: 00 (offset 11)
16:59:00.935 TRACE 3558: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: F6 (offset 11)
16:59:00.959 TRACE 3559: [emsesp] boiler(0x08) -B-> All(0x00), HPSilentMode(0x0484), data: F6 (offset 11)
```
[ems-esp-log-silent-temp.txt](https://github.com/user-attachments/files/17791278/ems-esp-log-silent-temp-30cut.txt)
![grafik](https://github.com/user-attachments/assets/eac7e606-a355-44bb-b84c-5ae30f3ae9b2)

**Comment by MichaelDvP** [comment]
The entities are all implemented. Are you using v3.7.x?
TimeOfDay formatting in Web is difficult, needs some refactoring. 
Also i don't know if it is possible in HA (i'm not using HA).
For mqtt publish i think a single number is better than a string.

**Reply by MarkusMit** [reply]
Using `3.7.0`. Sorry, I thought I had mentioned it somewhere...

**Comment by MarkusMit** [comment]
[Resolved] It seems something went wrong when updating last week. I did upload the `3.7.0` firmware, but it it was still showing a `3.7.0-dev` version I had installed months ago. 
Now I installed `3.7.1-dev.10` and now I also found the `powerreduction` entity.
